pkg/roachprod/install/cockroach.go

```diff
@@ -438,23 +439,31 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
 		// `--start-single-node` flag will handle all of this for us.
 		shouldInit := startOpts.Target == StartDefault && !c.useStartSingleNode() && !startOpts.SkipInit
 
-		for _, node := range c.Nodes {
-			// NB: if cockroach started successfully, we ignore the output as it is
-			// some harmless start messaging.
-			if err := c.startNode(ctx, l, node, startOpts); err != nil {
-				return err
-			}
-			// We reserve a few special operations (bootstrapping, and setting
-			// cluster settings) to the InitTarget.
-			if startOpts.GetInitTarget() != node {
-				continue
-			}
-
-			if shouldInit {
-				if err := c.initializeCluster(ctx, l, node); err != nil {
+		var g errgroup.Group
+		for _, n := range c.Nodes {
+			node := n
+			g.Go(func() error {
+				// NB: if cockroach started successfully, we ignore the output as it is
+				// some harmless start messaging.
+				if err := c.startNode(ctx, l, node, startOpts); err != nil {
```

Comment:
Maybe we should ensure the join node (i.e., `node == GetInitTarget`) is up before starting all others? There could be some pathological cases wherein the join node is delayed due to execution order (e.g., starvation). Also, if the join node can't be started, there is no point in attempting to start others.

Reply:
Done